encoder/vaapiencoder_hevc.cpp

```diff
@@ -0,0 +1,1648 @@
+
+    m_numSlices = 1;
+
+    assert (width() && height());
+
+    /* libva driver requred pic_width_in_luma_samples 16 aligned, not ctb aligned */
+    m_AlignedWidth = (width() + 15) / 16 * 16;
+    m_AlignedHeight = (height() + 15) / 16 * 16;
+
+    m_cuWidth = (width() + m_cuSize -1) / m_cuSize;
+    m_cuHeight = (height() + m_cuSize-1) / m_cuSize;
+
+    m_confWinLeftOffset = m_confWinTopOffset = 0;
+
+    if (m_AlignedWidth != width() || m_AlignedHeight !=height()) {
+            m_confWinFlag = true;
+            m_confWinRightOffset = (m_AlignedWidth - width()) / 2;
+            m_confWinBottomOffset = (m_AlignedHeight -height()) / 2;
+    } else {
+        m_confWinFlag = false;
+        m_confWinRightOffset = 0;
+        m_confWinBottomOffset = 0;
+    }
+
+    if (ipPeriod() == 0)
+        m_videoParamCommon.intraPeriod = 1;
+    else if (ipPeriod() >= 1)
+        m_numBFrames = ipPeriod() - 1;
+
+    assert(intraPeriod() > ipPeriod());
+
+    /* we set all I frame to be IDR frame right now */
+    keyFramePeriod() = intraPeriod();
+    if (keyFramePeriod() > MAX_IDR_PERIOD)
+        keyFramePeriod() = MAX_IDR_PERIOD;
+
+    if (minQP() > initQP() ||
+            (rateControlMode()== RATE_CONTROL_CQP && minQP() < initQP()))
+        minQP() = initQP();
+
+    if (m_numBFrames > (intraPeriod() + 1) / 2)
+        m_numBFrames = (intraPeriod() + 1) / 2;
+
+    DEBUG("resetParams, ensureCodedBufferSize");
+    ensureCodedBufferSize();
+
+    /* init m_maxFrameNum, max_poc */
+    m_log2MaxFrameNum =
+        hevc_get_log2_max_frame_num (keyFramePeriod());
+    assert (m_log2MaxFrameNum >= 4);
+    m_maxFrameNum = (1 << m_log2MaxFrameNum);
+    m_log2MaxPicOrderCnt = m_log2MaxFrameNum + 1;
+    m_maxPicOrderCnt = (1 << m_log2MaxPicOrderCnt);
+
+    m_maxRefList0Count = 1;
+    m_maxRefList1Count = m_numBFrames > 0;
+    m_maxRefFrames = m_maxRefList0Count + m_maxRefList1Count;
+
+    INFO("m_maxRefFrames: %d", m_maxRefFrames);
+
+    setShortRfs();
+
+    resetGopStart();
+}
+
+Encode_Status VaapiEncoderHEVC::getMaxOutSize(uint32_t *maxSize)
+{
+    FUNC_ENTER();
+
+    if (ensureCodedBufferSize())
+        *maxSize = m_maxCodedbufSize;
+    else
+        *maxSize = 0;
+
+    return ENCODE_SUCCESS;
+}
+
+Encode_Status VaapiEncoderHEVC::start()
+{
+    FUNC_ENTER();
+    resetParams();
+    return VaapiEncoderBase::start();
+}
+
+void VaapiEncoderHEVC::flush()
+{
+    FUNC_ENTER();
+    resetGopStart();
+    m_reorderFrameList.clear();
+    referenceListFree();
+    VaapiEncoderBase::flush();
+}
+
+Encode_Status VaapiEncoderHEVC::stop()
+{
+    flush();
+    return VaapiEncoderBase::stop();
+}
+
+Encode_Status VaapiEncoderHEVC::setParameters(VideoParamConfigType type, Yami_PTR videoEncParams)
+{
+    Encode_Status status = ENCODE_INVALID_PARAMS;
+    AutoLock locker(m_paramLock);
+
+    FUNC_ENTER();
+    if (!videoEncParams)
+        return ENCODE_INVALID_PARAMS;
+    switch (type) {
+    case VideoParamsTypeAVC: {
```

Comment:
could we typedef it to hevc? or rename it to VideoParamsType and VideoParams*. It may makes user confused.